src/qibo/tomography_RGD/qibo_states.py

```diff
@@ -0,0 +1,257 @@
+import random
+
+import numpy as np
+
+import qibo
+from qibo import gates, quantum_info
+
+
+class State:

```

Comment:
Correct me if I'm wrong, but it seems like this base class is unnecessary since every method is dependent on a `qibo.models.Circuit` method. I'd say that if these state classes are needed at all (which I'm not convinced of), you could use `Circuit` as the base class directly.